sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -46,6 +46,23 @@ class SaleOrderRecommendation(models.TransientModel):
     def _default_order_id(self):
         return self.env.context.get("active_id", False)
 
+    def _recomendable_sale_order_lines_domain(self):
+        """Domain to find recent SO lines."""
+        start = datetime.now() - timedelta(days=self.months * 30)
+        start = fields.Datetime.to_string(start)
+        other_sales = self.env["sale.order"].search([
+            ("partner_id", "child_of",
+             self.order_id.partner_id.commercial_partner_id.id),
+            ("date_order", ">=", start),
+        ])
+        return [
+            ("order_id", "!=", self.order_id.id),
```

Comment:
Don't do this for performance, but be sure `self.order_id` is not in `other_sales`

Reply:
I don't understand... What's the performance problem of excluding an ID? That's an indexed column and should be blazing fast... :confused:

Reply:
You are using 2 times `order_id`. Just do it once.